agent/06_agent_create_cluster.sh

```diff
@@ -163,13 +163,74 @@ function mce_complete_deployment() {
   mce_apply_postinstallation_manifests ${mceManifests}
 }
 
-create_image
-if [[ "${AGENT_DISABLE_AUTOMATED:-}" == "true" ]]; then
-  disable_automated_installation
-fi
+# Generate the PXE artifacts
+function create_pxe_files() {
+    local asset_dir=${1}
+    local openshift_install=${2}
+    "${openshift_install}" --dir="${asset_dir}" --log-level=debug agent create pxe-files
+}
+
+# Setup the environment to allow iPXE booting, by reusing libvirt native features
+# to configure dnsmaq tftp server and pxe boot file. Note that this is supported
+# only for IPv4.
+function setup_pxe_server() {
+    local PXE_SERVER_DIR=${WORKING_DIR}/pxe
+    mkdir -p ${PXE_SERVER_DIR}
+
+    # Create the iPXE boot file
+    local BASEURL=tftp://${PROVISIONING_HOST_EXTERNAL_IP}
+    sudo cat > ${PXE_SERVER_DIR}/boot.ipxe << EOF
+#!ipxe
+kernel ${BASEURL}/agent-vmlinuz.x86_64 initrd=main coreos.live.rootfs_url=${BASEURL}/agent-rootfs.x86_64.img ignition.firstboot ignition.platform.id=metal
+initrd --name main ${BASEURL}/agent-initrd.x86_64.img
+boot
+EOF
```

Comment:
Let's add the `ipxeBaseURL` in the agent-config.yaml [here](https://github.com/openshift-metal3/dev-scripts/blob/master/agent/roles/manifests/templates/agent-config_yaml.j2#L11) similar to [here](https://github.com/openshift/installer/pull/6723/files#diff-6adc75282b02154f0ac7fb942201d1f45026288a8e905cfb4cc9acd4cfb8cadfR37). The installer PR  to read the new field and generate the iPXE script is ready and is awaiting the merge https://github.com/openshift/installer/pull/6723 This way dev-scripts can also test the installer generating the iPXE script file and it can then be simply used here instead of creating it.

Reply:
Thanks @pawanpinjarkar it's a good idea we could follow up as soon as #6273 lands